### PR TITLE
Feature: Add blocking direction icons in cli

### PR DESCRIPTION
**Pull request by amovar18**
## Description
To increase feature parity between `extension` and the `CLI`, the blocking direction feature has been added to the CLI.

## Relevant Technical Choices
- Add network traffic details to the cookies.
- Derive the network blocking status from the network events.

## Testing Instructions
- Clone this branch. In the terminal run `npm run build-cli`.
- Now run `npm run cli https://livemint.com -- -nt`
- Post this open the HTML file generated by the CLI. In the table, you'll see the blocking direction signs.

## Screenshot/Screencast
Screenshot 2024-06-17 at 14 16 40

---

## Checklist

- [x] I have thoroughly tested this code to the best of my abilities.
- [x] I have reviewed the code myself before requesting a review.
- [x] This code is covered by unit tests to verify that it works as intended.
- [x] The QA of this PR is done by a member of the QA team (to be checked by QA).

Partially addresses #656

### CODE REVIEW

File: packages/common/src/utils/deriveBlockingStatus.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * Internal dependencies.
+ */
+import {
+  BLOCK_STATUS,
+  CookieData,
+  responsEvent,
```
Comment (ayushnirwal) [inline packages/common/src/utils/deriveBlockingStatus.ts:22]: Spelling Error

File: packages/cli-dashboard/src/components/utils/tests/reshapeCookies.ts [hunk 2]
```diff
@@ -31,7 +31,7 @@ describe('reshapeCookies', () => {
   it('should return an object with the cookies', () => {
     const cookies = {
       'https://edition.cnn.com': {
-        'countryCode:.cnn.com:/': {
+        'countryCode.cnn.com:/': {
```
Comment (ayushnirwal) [inline packages/cli-dashboard/src/components/utils/tests/reshapeCookies.ts:34]: Why did the key formating change?

File: packages/cli-dashboard/src/components/siteReport/tabs/cookies/cookiesListing/index.tsx [hunk 3]
```diff
@@ -86,6 +86,7 @@ const CookiesListing = ({
         className="h-full flex"
       >
         <CookieTable
+          //@ts-ignore
```
Comment (ayushnirwal) [inline packages/cli-dashboard/src/components/siteReport/tabs/cookies/cookiesListing/index.tsx:89]: Please add the reason for using '//@ts-ignore'.

File: packages/cli-dashboard/src/components/utils/tests/reshapeCookies.ts (hunk 2 above)
Comment (amovar18) [reply packages/cli-dashboard/src/components/utils/tests/reshapeCookies.ts:34]: It was to fix the test but that has been resolved now in latest commits.

File: packages/common/src/utils/deriveBlockingStatus.ts (hunk 1 above)
Comment (amovar18) [reply packages/common/src/utils/deriveBlockingStatus.ts:22]: Resolved in latest commit.

File: packages/cli-dashboard/src/components/siteReport/tabs/cookies/cookiesListing/index.tsx (hunk 3 above)
Comment (amovar18) [reply packages/cli-dashboard/src/components/siteReport/tabs/cookies/cookiesListing/index.tsx]: Resolved in latest commits.

**Review by mohdsayed — APPROVED** [review]

**Pull request merged**